tensornetwork/linalg/initialization.py

```diff
@@ -117,21 +117,22 @@ def ones_like(input: Union[np.ndarray,Tensor],
      backend(optional): The backend or its name."""
   if backend is None:
     backend = backend_contextmanager.get_default_backend()
-  elif isinstance(backend, str):
-    backend = backend_contextmanager.backend_factory.get_backend(backend)
   else:
-    backend = backend
-  if isinstance(input, Tensor): # encase input of type Tensor, create Tensor normally
+    backend = backend_contextmanager.backend_factory.get_backend(backend)
+  if isinstance(input, Tensor):  # incase input of type Tensor, create Tensor normally
     the_tensor = initialize_tensor("ones", input.shape, backend=input.backend, dtype=input.dtype)
     return the_tensor
-  if isinstance(input, np.ndarray): # encase input of type np.ndarray, convert to Tensor
-    input = convert_to_tensor(input,dtype=dtype)
-    the_tensor = initialize_tensor("ones", input.shape, backend=backend, dtype=dtype)
+  else: # incase input of type np.ndarray, convert to Tensor
+    try:
```

Comment:
```suggestion
    try:
      input = backend.convert_to_tensor(input)
    except TypeError as e:
      errstr = "Input to ones_like had invalid type, causing error message: \n" +str(e)
      raise TypeError(errstr)
```

Reply:
There's a very confusing situation going on right now within the codebase where, for legacy reasons, the word `Tensor' is used to refer to two very different things. At the level of tn.Tensor, it refers to the tn.Tensor object. Within the backends, it refers to the array class of the backend. Thus `convert_to_tensor` does not convert to `tn.Tensor`, but rather to the array class of the relevant backend.

Reply:
also raise the last TypeError with a `from`:
```python 
raise TypeError(errstr) from e

```